qiita_db/analysis.py

```diff
@@ -204,15 +205,20 @@ def delete(cls, _id):
             if not cls.exists(_id):
                 raise qdb.exceptions.QiitaDBUnknownIDError(_id, "analysis")
 
+            # Check if the analysis has any artifact
+            sql = """SELECT EXISTS(SELECT *
+                                   FROM qiita.analysis_artifact
+                                   WHERE analysis_id = %s)"""
+            qdb.sql_connection.TRN.add(sql, [_id])
+            if qdb.sql_connection.TRN.execute_fetchlast():
+                raise qdb.exceptions.QiitaDBOperationNotPermittedError(
+                    "Can't delete analysis %d, has artifacts attached")
```

Comment:
This format string is missing the analysis identifier, can you add the appropriate formatting?